src/main/kotlin/lotto/model/Lotto.kt

```diff
@@ -1,35 +1,46 @@
 package lotto.model
 
-data class Lotto(val lottoNumbers: List<LottoNumber>) {
+data class Lotto(private val lottoNumbers: Set<LottoNumber>) {
+    var lottoNums: Set<LottoNumber>
+        private set
+
     init {
         require(isUniqueSixNumbers(lottoNumbers)) { "로또 번호는 서로 다른 6개의 숫자여야만 합니다." }
+        lottoNums = lottoNumbers
     }
 
-    constructor(numbers: Collection<Int>) : this(numbers.map(::LottoNumber))
+    constructor(numbers: Collection<Int>) : this(numbers.map(::LottoNumber).toSet())
+
+    constructor(stringNumbers: String) : this(Delimiter(stringNumbers).split().map { it.toInt() })
 
-    constructor(stringNumbers: String) : this(splitByComma(stringNumbers))
+    fun getWinningCount(winningNumbers: Lotto): Int {
+        return lottoNums.count { winningNumbers.lottoNums.contains(it) }
+    }
 
-    fun getWinningCount(winningNumbers: List<Int>): Int {
-        return lottoNumbers.filter { winningNumbers.contains(it.number) }.count()
+    fun hasBonusNumber(bonusNumber: LottoNumber): Boolean {
+        return lottoNums.contains(bonusNumber)
+    }
+
+    fun getResult(winningLotto: Lotto, bonusNumber: LottoNumber): Coincidence? {
+        return Coincidence.values()
+            .find { it.coincidenceCount == getWinningCount(winningLotto) && it.hasBonusNum == hasBonusNumber(bonusNumber) }
     }
 
     override fun toString(): String {
-        return lottoNumbers.map { it.number }.joinToString { ", " }
+        return lottoNumbers
+            .sortedBy { it.number }
+            .map { it.number }
```

Comment:
아래의 코드로 개선해 보면 어떨까요?

```suggestion
            .map { it.number }
            .sorted()
```